src/main/java/org/kairosdb/core/http/rest/metrics/DefaultQueryMeasurementProvider.java

```diff
@@ -115,15 +105,11 @@ private void measureSpan(final Histogram histogram, final QueryMetric query) {
 	}
 
 	private void measureDistance(final Histogram histogram, final QueryMetric query) {
-		measureInternal(query.getStartTime(), Long.MAX_VALUE, histogram, "query_distance_in_days");
+		measureInternal(query.getStartTime(), System.currentTimeMillis(), histogram, "query_distance_in_days");
 	}
 
 	private void measureInternal(final long startTime, final long endTime, final Histogram histogram, final String tag) {
-		final long now = System.currentTimeMillis();
-		// ensure that: now-TTL <= startTime <= now, and startTome <= endTime <= now
-		final long actualStartTime = Math.max(Math.min(startTime, now), now - datapoints_ttl * 1000);
-		final long actualEndTime = Math.max(Math.min(endTime, now), actualStartTime);
-		final long timeInMillis = actualEndTime - actualStartTime;
+		final long timeInMillis = endTime - startTime;
```

Comment:
Are we sure that this point that `endTime >= startTime`?

Reply:
yes, because [QueryParser throws exception if not](https://github.com/zalando-zmon/kairosdb/blob/zalando-kairosdb/src/main/java/org/kairosdb/core/http/rest/json/QueryParser.java#L219) and then [the statistics isn't measures](https://github.com/zalando-zmon/kairosdb/blob/zalando-kairosdb/src/main/java/org/kairosdb/core/http/rest/MetricsResource.java#L409-L444)